client/src/main/java/com/alibaba/nacos/client/config/impl/ClientWorker.java

```diff
@@ -775,50 +775,35 @@ public void notifyListenConfig() {
         
         @Override
         public void executeConfigListen() {
-            
             Map<String, List<CacheData>> listenCachesMap = new HashMap<>(16);
             Map<String, List<CacheData>> removeListenCachesMap = new HashMap<>(16);
             long now = System.currentTimeMillis();
             boolean needAllSync = now - lastAllSyncTime >= ALL_SYNC_INTERNAL;
+            
             for (CacheData cache : cacheMap.get().values()) {
-                
                 synchronized (cache) {
-                    
                     //check local listeners consistent.
                     if (cache.isConsistentWithServer()) {
                         cache.checkListenerMd5();
                         if (!needAllSync) {
                             continue;
                         }
                     }
-                    
-                    if (!cache.isDiscard()) {
```

Comment:
Why remove this judgement?

Reply:
I have not removed this judgement; I just moved it inside the if (!cache.isUseLocalConfigInfo()) { judgement, because regardless of the result of if (!cache.isDiscard()) {, the if (!cache.isUseLocalConfigInfo()) { judgement will be executed